apps/condo/domains/common/components/trackers/TrackerInstance.ts

```diff
@@ -0,0 +1,104 @@
+import getConfig from 'next/config'
+import { get, compact } from 'lodash'
+import { validate as uuidValidate } from 'uuid'
+
+export type ITrackerLogEventType = {
+    eventName: string
+    eventProperties?: Record<string, unknown>
+    denyDuplicates?: boolean
+}
+
+/**
+ * Abstract class representing a tracking instance library, such as amplitude.js
+ * @abstract
+ * @class TrackerInstance
+ * @classdesc Should be implemented when new tracking library would be connected
+ * @param {string} instanceName string representation of tracker key inside environment runtime config
+ */
+abstract class TrackerInstance {
+    protected readonly token: string | null = null
+    protected readonly configParams: Record<string, unknown> = {}
+    protected readonly allowedDomains: Record<string, unknown> = {}
+    protected readonly instanceName: string
+    protected instance = null
+    private prevEvent: string
+
+    protected constructor (instanceName: string, localConfig?: Record<string, unknown>) {
+        this.instanceName = instanceName
+        const config = localConfig ? localConfig : getConfig()
+        const trackingConfig = get(config, 'publicRuntimeConfig.trackingConfig')
+
+        if (trackingConfig) {
+            const token = get(trackingConfig, ['trackers', this.instanceName, 'token'])
+            const configParams = get(trackingConfig, ['trackers', this.instanceName, 'config'])
+            if (!token) {
+                console.error(`No token for ${this.instanceName} tracker instance was provided!`)
+            }
+
+            this.token = token
+            this.configParams = configParams
+            this.allowedDomains = get(trackingConfig, 'domains', {})
+        }
+    }
+
+    /**
+     * This method is called when the component is created.
+     * Should be wrapped at window object exists condition if it is necessary!
+     * @abstract
+     * @return void
+     */
+    abstract init (): void
+
+    /**
+     * Common method for sending data to tracker API
+     * @abstract
+     * @param {string} eventName - short event description
+     * @param {Object} eventProperties - related properties
+     * @param {Boolean} denyDuplicates - enable check for duplicated events on eventName prop comparison
+     * @return void
+     */
+    abstract logEvent ({ eventName, eventProperties, denyDuplicates }: ITrackerLogEventType): void
```

Comment:
In such cases it's more common way it to use private abstract function like `_logEvent` and public `logEvent` which call the `isNeedToSendEvent` before the `_logEvent`.

You need to keep public API stable and controllable. If you give an opportunity to any Tracke to realize their own contract you really don't need `isNeedToSendEvent` here.